### ISSUE TITLE
Regarding when the `package-name` used in the recipe format can be omitted.

### ISSUE THREAD

**Issue by hongyi-zhao** [raxod502/straight.el#828]
As described [here](https://github.com/raxod502/straight.el#the-recipe-format), the general format for a `straight.el` recipe is:

>```emacs-lisp
>(package-name :keyword value :keyword value ...)
>```
And according to the further detailed comments [here](https://github.com/raxod502/straight.el#my-use-package-form-isnt-working-properly): 

> If the package name is different from the feature name, then pass `:straight <package-name>`.

So I think that, as far as the previously discussed problem [here](https://github.com/raxod502/straight.el/issues/811) is concerned, I further discuss and summarize the corresponding `straight.el` recipes as the following:

```emacs-lisp
(use-package elisp
  :straight (zenitani-elisp :local-repo "zenitani-elisp" :type git :host github :repo "zenitani/elisp"))
```

The above code snippet will be expanded to the following by `straight.el`:

```emacs-lisp
(straight-use-package
 '(zenitani-elisp :local-repo "zenitani-elisp" :type git :host github :repo "zenitani/elisp"))
```
With the above usage, the build path of this package will locate at `~/.emacs.d/straight/build/zenitani-elisp`, and the corresponding `autoloads.el` file, i.e., `~/.emacs.d/straight/build/zenitani-elisp-autoloads.el`, will be created by `straight.el`.

OTOH, if we use the following recipe:

```emacs-lisp
(use-package elisp
  :straight (:local-repo "zenitani-elisp" :type git :host github :repo "zenitani/elisp"))
```
The above code snippet will be expanded to the following by `straight.el`:
```emacs-lisp
(straight-use-package
   '(elisp :local-repo "zenitani-elisp" :type git :host github :repo "zenitani/elisp"))
```

In this case, if the feature name `elisp` exists, it will work as it is, otherwise, an error similar to the following will be triggered:

> Error (use-package): Cannot load `elisp`.

To summarize: the first argument after `use-package` must be [a named feature](https://www.gnu.org/software/emacs/manual/html_node/elisp/Named-Features.html); while the `<package-name>` passed to `straight.el` just be treated as a convenient trick for determining the package's name under `straight.el`'s `build` directory, and subsequently also be used as the prefix name of the `autoloads.el` file automatically created by `straight.el` for this package.

Regards,
BY
